docs/proposals/nginx-extensions.md

```diff
@@ -0,0 +1,964 @@
+| Active health checks                                                                             | X                   |
+| JSON Web Token (JWT) authentication                                                              | X                   |
+| OpenID Connect (OIDC) authentication. Allow users to authenticate with single-sign-on (SSO)      | X                   |
+| Customize client settings. For example, `client_max_body_size`                                   |                     |
+| Upstream keepalives                                                                              |                     |
+| Client keepalives                                                                                |                     |
+| TLS settings. For example, TLS protocols and server ciphers                                      |                     |
+| OpenTelemetry tracing                                                                            |                     |
+
+### Medium-Priority Features
+
+| Features                                                                               | Requires NGINX Plus |
+|----------------------------------------------------------------------------------------|---------------------|
+| Circuit breaker                                                                        |                     |
+| Load-balancing method                                                                  |                     |
+| Load-balancing method (least time)                                                     | X                   |
+| Limit connections to a server                                                          |                     |
+| Request queuing. Queue requests if servers are unavailable                             | X                   |
+| Basic authentication. User/Password                                                    |                     |
+| PROXY protocol                                                                         |                     |
+| Upstream zone size                                                                     |                     |
+| Custom response. Return a custom response for a given path                             |                     |
+| External routing. Route to services outside the cluster                                |                     |
+| HTTP Strict Transport Security (HSTS)                                                  |                     |
+| API Key authentication                                                                 |                     |
+| Next upstream retries. NGINX retries by trying the _next_ server in the upstream group |                     |
+| Proxy Buffering                                                                        |                     |
+| Connection timeouts                                                                    |                     |
```

Comment:
timeouts and retries are stability-related configuration of backends, similar to passive and active health checks. I think it makes sense to move them to high-priority because of that.
more over, timeouts (incompatible with NGINX) were already added to Gateway API (experimental channel) and retries are being worked on -- to gauge their popularity/importance.

Reply:
@mpstefan thoughts on this?

@pleshakov should we prioritize NGINX-specific timeouts/retries or prioritize implementing the Gateway API's timeouts/retires (likely with custom modules)? I think we should prioritize the latter, especially since timeouts are already in the experimental channel.

Reply:
> @pleshakov should we prioritize NGINX-specific timeouts/retries or prioritize implementing the Gateway API's timeouts/retires (likely with custom modules)? I think we should prioritize the latter, especially since timeouts are already in the experimental channel.

those connection timeouts are still useful and they are already available to configure.
with retries, I don't see big difference between NGINX and what is asked here https://github.com/kubernetes-sigs/gateway-api/issues/1731 . But at the same time, there is no yet GEP proposed.